### ISSUE TITLE
Evaluation loss issue (strange trend)

### ISSUE THREAD

**Issue by FHL1998** [artidoro/qlora#150]
A wierd thing happen when I insepct the eval_loss, it decrease and then increase in a strange way. Anyone have idea of solving it?
![1686276272335](https://github.com/artidoro/qlora/assets/59732331/6fb6a635-808e-4fcd-8f70-a082e24826bf)

**Comment by jocastrocUnal** [comment]
same here 
![image](https://github.com/artidoro/qlora/assets/93233286/57835276-e127-4e86-9421-dc801de57361)

**Comment by artidoro** [comment]
Validation loss will often decrease and then increase later. In deep learning people often train past the point where validation loss starts increasing even though that indicates some "overfitting" to the training set. I would recommend tracking performance on the actual downstream metrics you are interested in improving throughout training.

**Comment by OfirArviv** [comment]
I have the same problem. I fine-tuned mt5-xxl, ul2 and xglm-7.5 model on 2 datasets, and the model manage to "learn" for a good amount of steps, but usually after an epoch or 2 the model starts to diverge, the loss increase and also the task metric itself. It's get to the 80% accuracy and then quickly decrease to 0%. This behavior is consistent numerous  several runs.

The same code exactly without the 4bit part (only Lora) behave normally - the model does not diverge even after 50 epochs.
So it seem to me it has something to do with the training dynamic. I'm trying now different lr to see if that has something to do with that.
